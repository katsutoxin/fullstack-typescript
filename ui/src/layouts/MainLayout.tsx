import * as React from 'react';
import * as styles from './MainLayout.scss';
import {SPC} from 'lib/SPC';

const logo = require('../img/logo.svg');

export class MainLayout extends SPC<{}> {
	render() {
		return (
			<div className={styles.mainLayout}>
				<div className={styles.header}>
					<img src={logo} className={styles.logo} alt='logo'/>
					<h2>Welcome to Todos</h2>
				</div>
				{this.props.children}
			</div>
		);
	}
}
