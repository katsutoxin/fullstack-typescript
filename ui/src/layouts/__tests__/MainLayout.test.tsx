import * as React from 'react';
import {createRenderer} from 'react-test-renderer/shallow';
import {MainLayout} from '../MainLayout';

const renderer = createRenderer();

describe('MainLayout', () => {

	it('renders correctly', () => {
		renderer.render(<MainLayout><div id='content'/></MainLayout>);

		expect(renderer.getRenderOutput()).toMatchSnapshot();
	});
});
