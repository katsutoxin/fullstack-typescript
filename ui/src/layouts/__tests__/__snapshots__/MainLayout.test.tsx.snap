exports[`MainLayout renders correctly 1`] = `
<div
  className="mainLayout">
  <div
    className="header">
    <img
      alt="logo"
      className="logo"
      src="logo.svg" />
    <h2>
      Welcome to Todos
    </h2>
  </div>
  <div
    id="content" />
</div>
`;
